src/imputation/MoR.py

```diff
@@ -112,8 +121,108 @@ def carry_forwards(df, backdata, impute_vars):
         df.loc[match_cond, f"{var}_imputed"] = df.loc[match_cond, f"{var}_prev"]
     df.loc[match_cond, "imp_marker"] = "CF"
 
+    df.loc[match_cond] = create_imp_class_col(df,
+                                              "200_prev",
+                                              "201_prev",
+                                              use_cellno=False)
+
     # Drop merge related columns
-    to_drop = [column for column in df.columns if column.endswith("_prev")]
+    to_drop = [column for column in df.columns
+               if (column.endswith("_prev")) & (column not in impute_vars)]
     to_drop += ["_merge"]
     df = df.drop(to_drop, axis=1)
     return df
+
+
+def calculate_links(current_df, prev_df, target_vars):
+    """Calculate the links between previous and current data.
+
+    Args:
+        current_df (_type_): _description_
+        prev_df (_type_): _description_
+        target_vars (_type_): _description_
+    """
+    link_df = pd.merge(current_df,
+                       prev_df,
+                       on=["reference", "imp_class"],
+                       how="inner",
+                       suffixes=("", "_prev"))
+    target_vars_prev = [f"{var}_prev" for var in target_vars]
+    link_df = (link_df[["reference", "imp_class"] + target_vars + target_vars_prev]
+               .groupby(["reference", "imp_class"])
+               .sum()
+               )
```

Comment:
looks good. Do you need to do a reset_index()?

Reply:
We already do it on line 160, but I guess it makes more sense to do it here given it's the last operation using the groupby.

Reply:
Have updated this now - weren't selecting the columns to keep properly, but have added some regex to do this.